pkg/auth/manager.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/maps/ctmap"
+	"github.com/cilium/cilium/pkg/monitor"
+	"github.com/cilium/cilium/pkg/u8proto"
+)
+
+type AuthManager struct {
+	endpointManager *endpointmanager.EndpointManager
+}
+
+func NewAuthManager(epMgr *endpointmanager.EndpointManager) *AuthManager {
+	return &AuthManager{
+		endpointManager: epMgr,
+	}
+}
+
+func (a *AuthManager) AuthRequired(dn *monitor.DropNotify, ci *monitor.ConnectionInfo) {
+	log.Infof("policy: Authentication required for identity %d->%d, %s %s:%d->%s:%d",
+		dn.SrcLabel, dn.DstLabel, ci.Proto, ci.SrcIP, ci.SrcPort, ci.DstIP, ci.DstPort)
+
+	/* Find the local endpoint. */
+	ingress := false
+	localAddr := ci.SrcIP.String()
+	remoteAddr := ci.DstIP.String()
+	if dn.DstID != 0 {
+		ingress = true
+		/* swap addresses for ingress */
+		addr := localAddr
+		localAddr = remoteAddr
+		remoteAddr = addr
+	}
+	proto, err := u8proto.ParseProtocol(ci.Proto)
+	if err != nil {
+		log.Warningf("auth: Cannot parse protocol %q: %v", ci.Proto, err)
+	}
+	ep := a.endpointManager.LookupCiliumID(dn.Source)
+	/* Update CT to mask as authorized. */
+
+	err = ctmap.Update(ep.ConntrackName(), remoteAddr, localAddr, proto, ingress,
```

Comment:
Come to think of if we may need to pass the auth type enum to the datapath and back up so that we can call the corresponding auth handler from here. We could take a pad byte in policy entry for this purpose, replacing the flag added in this PR? If we can pass that byte back in the perf event, then we would not need to add the same info to the CT entry (that also does not have pad bytes left)?

Reply:
Added passing in the auth type to the policy map and then back to userspace in the drop notification.